File: src/models/donationType.model.js
const mongoose = require('mongoose');

const donationTypeSchema = new mongoose.Schema(
  {
    title: { type: String, required: true, trim: true },
    category: {
      type: String,
      enum: ['regular', 'zakat', 'emergency'],
      required: true,
    },
    amount: { type: Number, required: true, min: 0 },
    description: { type: String, default: '', trim: true },
    isActive: { type: Boolean, default: true },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model('DonationType', donationTypeSchema);

File: AICOM_App-Frontend/lib/Models/donation/donation_type_model.dart
class DonationType {
  final String id;
  final String title;
  final String category; // regular, zakat, emergency
  final double amount;
  final String description;
  final bool isActive;

  DonationType({
    required this.id,
    required this.title,
    required this.category,
    required this.amount,
    required this.description,
    required this.isActive,
  });

  factory DonationType.fromJson(Map<String, dynamic> json) {
    return DonationType(
      id: json['_id'] ?? json['id'] ?? '',
      title: json['title'] ?? '',
      category: json['category'] ?? 'regular',
      amount: (json['amount'] ?? 0).toDouble(),
      description: json['description'] ?? '',
      isActive: json['isActive'] ?? true,
    );
  }
}
